### PR TITLE
[WFLY-12243]: Monitor resource adapter thread pools.

### PR DESCRIPTION
**Pull request by ehsavoie**
* Expose metrics on global client thread pool and scheduled thread pool.

Jira: https://issues.jboss.org/browse/WFLY-12243

### CODE REVIEW

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+:idprefix:          messaging,jms
+:idseparator:       -
+
+== Overview
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE[WFCORE-12243]
```
Comment (jmesnil) [inline messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: text and link are not correct.

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+* `completed-task-count`: The approximate number of tasks that have been executed by all ActiveMQ clients.
+* `current-thread-count`: The current number of threads in the pool used by all ActiveMQ clients.
+* `largest-thread-count`: The largest number of threads that have ever simultaneously been in the pool used by all ActiveMQ clients.
+* `task-count`: The approximate total number of tasks that have ever been scheduled by all ActiveMQ clients.
+
+
+=== Hard Requirements
+
+* Add a integer `global-client-scheduled-thread-pool-active-count` metric to https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/[/subsystem=messaging-activemq]
+* Add a long `global-client-scheduled-thread-pool-completed-task-coun` metric to https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/[/subsystem=messaging-activemq]
```
Comment (jmesnil) [inline messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: typo: `global-client-scheduled-thread-pool-completed-task-count`

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+* Add a integer `global-client-scheduled-thread-pool--current-thread-count` metric to https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/[/subsystem=messaging-activemq]
```
Comment (jmesnil) [inline messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: typo: `global-client-scheduled-thread-pool-current-thread-count`

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+* Add a long (in nanoseconds) `global-client-scheduled-thread-pool-keepalive-time` metric to https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/[/subsystem=messaging-activemq]
+* Add a integer `global-client-scheduled-thread-pool-largest-thread-count` metric to https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/[/subsystem=messaging-activemq]
+* Add a long `global-client-scheduled-thread-pool-task-count` metric to https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/[/subsystem=messaging-activemq]
+* Add a integer `global-client-thread-pool-active-count` metric to https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/[/subsystem=messaging-activemq]
+* Add a long `global-client-thread-pool-completed-task-coun` metric to https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/[/subsystem=messaging-activemq]
```
Comment (jmesnil) [inline messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: typo: `global-client-scheduled-thread-pool-current-thread-count`

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+* Add a integer `global-client-thread-pool--current-thread-count` metric to https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/[/subsystem=messaging-activemq]
```
Comment (jmesnil) [inline messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: typo: `global-client-scheduled-thread-pool-current-thread-count`

**Comment by jmesnil** [comment]
other than the typos, looks good to me

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+[standalone@localhost:9990 /] /subsystem=messaging-activemq:read-resource(include-runtime)
+{
+    "outcome" => "success",
+    "result" => {
+        "global-client-scheduled-thread-pool-active-count" => 0,
+        "global-client-scheduled-thread-pool-completed-task-count" => 0L,
+        "global-client-scheduled-thread-pool-current-thread-count" => 0,
+        "global-client-scheduled-thread-pool-keepalive-time" => 10000000L,
+        "global-client-scheduled-thread-pool-largest-thread-count" => 0,
+        "global-client-scheduled-thread-pool-max-size" => undefined,
```
Comment (istraka) [inline messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: I do not see thread-pool-max-size in requirements. Has it been already implemented?

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        "global-client-scheduled-thread-pool-task-count" => 0L,
+        "global-client-thread-pool-active-count" => 0,
+        "global-client-thread-pool-completed-task-count" => 2L,
+        "global-client-thread-pool-current-thread-count" => 2,
+        "global-client-thread-pool-keepalive-time" => 60000000000L,
+        "global-client-thread-pool-largest-thread-count" => 2,
+        "global-client-thread-pool-max-size" => undefined,
```
Comment (istraka) [inline messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: I do not see thread-pool-max-size in requirements. Has it been already implemented?

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+// Discuss with QE during the Kickoff state to decide this
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+== Requirements
```
Comment (istraka) [inline messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: `core-size` and `max-size` metrics are missing, aren't they? They are in the description of https://issues.jboss.org/browse/EAP7-1089.

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc (hunk 7 above)
Comment (ehsavoie) [reply messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: global-client-scheduled-thread-pool-max-size is already existing and is a configuration parameter to define the max pool size

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc (hunk 8 above)
Comment (ehsavoie) [reply messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: No they are not metrics ;)

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc (hunk 6 above)
Comment (ehsavoie) [reply messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: global-client-scheduled-thread-pool-max-size is already existing and is a configuration parameter to define the max pool size

File: messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc (hunk 8 above)
Comment (istraka) [reply messaging/WFLY-12243_monitor_artemis_resource_adapter_thread_pools.adoc]: Agree. And keep-alive-time?

**Pull request merged**